src/main/java/org/jabref/gui/groups/GroupNodeViewModel.java

```diff
@@ -184,10 +188,11 @@ private void calculateNumberOfMatches() {
         // We calculate the new hit value
         // We could be more intelligent and try to figure out the new number of hits based on the entry change
         // for example, a previously matched entry gets removed -> hits = hits - 1
-        new Thread(()-> {
-            int newHits = groupNode.calculateNumberOfMatches(databaseContext.getDatabase());
-            Platform.runLater(() -> hits.setValue(newHits));
-        }).start();
+        BackgroundTask<Integer> calculateHits = BackgroundTask
+                .run(() -> groupNode.calculateNumberOfMatches(databaseContext.getDatabase()))
```

Comment:
This seems little, but is very important: Please rename the method `run()` in `BackgroundTask` to something else, suggestion: `wrap()`. `run()` has a very specific meaning in parallel Java code and by just looking at this code here, I thought: "This code runs a task not in parallel, but on the calling thread". And this is not at all what this code does. It just wraps a `Callable` as a `BackgroundTask`, execution comes later. The name should reflect that.